tdrs-backend/tdpservice/parsers/util.py

```diff
@@ -255,7 +259,32 @@ def parse_and_validate(self, line, generate_error):
         records = []
 
         for schema in self.schemas:
-            r = schema.parse_and_validate(line, generate_error)
-            records.append(r)
+            record, is_valid, errors = schema.parse_and_validate(line, generate_error)
+            records.append((record, is_valid, errors))
 
         return records
+
+
+def begin_transaction():
+    """Create manual database transaction."""
+    transaction.set_autocommit(False)
```

Comment:
these feel a little funny to me because they just use an imported `db.transaction`. i would expect some sort of transaction per-request, that you'd have to pass into these util functions. instead, it seems like there's one long-running transaction where you turn `autocommit` on/off. am i understanding that correctly? and are there any implications to this that you're aware of?